src/components/Compose/ComposeSend/useSendLens.ts

```
@@ -4,22 +4,60 @@ import { useCallback } from 'react';
 import { commentPostForLens, publishPostForLens, quotePostForLens } from '@/helpers/publishPost.js';
 import { useCustomSnackbar } from '@/hooks/useCustomSnackbar.js';
 import { ComposeModalRef } from '@/modals/controls.js';
+import { uploadFileToIPFS } from '@/services/uploadToIPFS.js';
 import { useComposeStateStore } from '@/store/useComposeStore.js';
 import { useLensStateStore } from '@/store/useLensStore.js';
+import type { MediaObject } from '@/types/index.js';
 
 export function useSendLens() {
     const currentProfile = useLensStateStore.use.currentProfile();
-    const { type, post, chars, images, video } = useComposeStateStore();
+    const { type, post, chars, images, updateImageByIndex, video, updateVideo, updateLensPostId } =
+        useComposeStateStore();
     const enqueueSnackbar = useCustomSnackbar();
 
     return useCallback(async () => {
         if (!currentProfile?.profileId) return;
+        const uploadedImages = await Promise.all(
+            images.map(async (media, index) => {
+                if (media.ipfs) return media;
+                const response = await uploadFileToIPFS(media.file);
+                if (response) {
+                    const patchedMedia: MediaObject = {
+                        ...media,
+                        ipfs: response,
+                    };
+                    // TODO race conditions
+                    updateImageByIndex(index, patchedMedia);
+                    return patchedMedia;
+                }
```

If failed to upload a image, should throw an error here.